tough/src/schema/mod.rs

```
@@ -120,6 +123,18 @@ impl Root {
             keys: &self.keys,
         }
     }
+
+    /// Given an object/key that impls Sign, return the corresponding
+    /// key ID from Root
+    #[allow(clippy::unnecessary_operation)]
+    pub fn key_id(&self, key_pair: &dyn Sign) -> Option<Decoded<Hex>> {
+        for (key_id, key) in &self.keys {
+            if key_pair.tuf_key() == *key {
+                Some(key_id.clone());
+            }
+        }
+        None
```

Is this clippy warning because we're not actually `return`ing the cloned `key_id`?  It seems like it'll always return `None`.

---

:)

Yes - that little `return` fixed all my testing issues as well. :)

---

Clippy can be pretty helpful at times.
Another one I've found is if I'm calling `fn do_something() -> Result<()>` like this `do_something()` clippy helpfully tells me I'm ignoring the error!